src/Google.Api.Gax/ClientHelper.cs

```diff
@@ -54,33 +55,51 @@ public ClientHelper(ServiceSettingsBase settings)
+                    headers: _clientCallSettings.Headers, // TODO: Add GAX header(s)
+                    deadline: CalculateDeadline(_clock, _clientCallSettings.Expiration),
+                    cancellationToken: _clientCallSettings.CancellationToken ?? default(CancellationToken),
+                    writeOptions: _clientCallSettings.WriteOptions,
+                    propagationToken: _clientCallSettings.PropagationToken,
+                    credentials: _clientCallSettings.Credentials);
             }
             return new CallOptions(
                 // TODO: Sort out our cloning story.
-                headers: callSettings.Headers ?? _globalCallSettings.Headers, // TODO: Add GAX header(s)
-                deadline: CalculateDeadline(_clock, callSettings.Expiration ?? _globalCallSettings.Expiration),
-                cancellationToken: cancellationToken ?? callSettings.CancellationToken ?? _globalCallSettings.CancellationToken ?? default(CancellationToken),
-                writeOptions: callSettings.WriteOptions ?? _globalCallSettings.WriteOptions,
-                propagationToken: callSettings.PropagationToken ?? _globalCallSettings.PropagationToken,
-                credentials: callSettings.Credentials ?? _globalCallSettings.Credentials);
+                headers: callSettingsOverride.Headers ?? _clientCallSettings.Headers, // TODO: Add GAX header(s)
+                deadline: CalculateDeadline(_clock, callSettingsOverride.Expiration ?? _clientCallSettings.Expiration),
+                cancellationToken: callSettingsOverride.CancellationToken ?? _clientCallSettings.CancellationToken ?? default(CancellationToken),
+                writeOptions: callSettingsOverride.WriteOptions ?? _clientCallSettings.WriteOptions,
+                propagationToken: callSettingsOverride.PropagationToken ?? _clientCallSettings.PropagationToken,
+                credentials: callSettingsOverride.Credentials ?? _clientCallSettings.Credentials);
+        }
+
+        /// <summary>
+        /// Builds an <see cref="ApiCall"/> given suitable underlying async and sync calls.
+        /// </summary>
+        /// <typeparam name="TRequest">Request type, which must be a protobuf message.</typeparam>
+        /// <typeparam name="TResponse">Response type, which must be a protobuf message.</typeparam>
+        /// <param name="asyncGrpcCall">The underlying synchronous gRPC call.</param>
+        /// <param name="syncGrpcCall">The underlying asynchronous gRPC call.</param>
+        /// <returns></returns>
+        public ApiCall<TRequest, TResponse> BuildApiCall<TRequest, TResponse>(
+            Func<TRequest, CallOptions, AsyncUnaryCall<TResponse>> asyncGrpcCall,
+            Func<TRequest, CallOptions, TResponse> syncGrpcCall)
+            where TRequest : class, IMessage<TRequest>
+            where TResponse : class, IMessage<TResponse>
+        {
+            ApiCall<TRequest, TResponse>.AsyncCall asyncCall = (request, callSettings) =>
```

Comment:
We might consider making BuildCallOptions a static call, and hoisting the _clientCallSettings variable into a local variable here, to make the lambda expression capture more efficient. Not just yet though :)